src/pages/signup.tsx

```diff
@@ -0,0 +1,335 @@
+import React, {useState, useEffect, Component} from 'react';
+import {useRouter} from 'next/router';
+
+import {Navbar, Page, Footer, GamerCard} from './../components/AppComponents';
+import {
+    Container,
+    Header,
+    CardBody,
+    Card,
+    Alert,
+    CardFooter,
+    Button,
+    Paragraph,
+    CardHeader,
+    Box,
+    Text,
+    Small,
+    Main,
+    LineBreak, UnorderedList, ListItem
+} from './../components/SimpleComponents';
+import {Input} from './../components/SmartComponents';
+
+import CONSTANTS from './../config/CONSTANTS';
+
+import {UserService} from './../components/Users';
+import UtilityService, {getBaseUrlWithProtocol} from '../services/UtilityService';
+import TypeService from '../services/TypeService';
+
+//===----=---=-=--=--===--=-===----=---=-=--=--===--=-===----=---=-=--=--===--=-//
+
+
+let SignUpPage = ({baseUrl}) => {
+    const MAX_LENGTH = 130;
+    const MIN_PASSWORD_LENGTH = 8;
+
+    function sanitizeBasicInput(value) {
+        value = UtilityService.validateItem(value, 'string', '').trim();
+
+        if (value.length > MAX_LENGTH) {
+            value.length = MAX_LENGTH;
+        }
+
+        return value;
+    }
+
+    const FORM_INPUT_CONFIG = {
```

Comment:
Form validation list items. This lets us validate and sanitize inputs more easily than having a master function with the different validation methods for each